src/test/geometries/CylinderTests.java

```diff
@@ -0,0 +1,29 @@
+package geometries;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import primitives.NormalizedVector;
+import primitives.Point;
+import primitives.Ray;
+import primitives.Vector;
+
+public class CylinderTests {
+	public final Ray ray = new Ray(new Point(3,2,1), new Vector(6,5,4));
+	public final Cylinder base = new Cylinder(ray, 5, 10);
+
+	@Test 
+	public void normal() {
```

Comment:
1) Like on the corner where the flat part meets the curve?
2) Why could we expect a difference at the center of the disk?

Reply:
1. Yes
2. Idk it's an edge case. We're not supposed to take the implementation into account. Tho I guess it's the same as any other point on the disk, so idk. Do what u think for this one

Reply:
1. I should just check that the normal is either correct wrt to the curved side or wrt the flat side? (Implementation wise I think it will always be considered on the flat side, since once you're within the margin of error it will decide that the plane contains() the point.)

Reply:
1. There's no need to enumerate your questions if there's only one.
2. In my original comment I already said what you should do.
3. You should check both with an OR